core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -750,10 +780,18 @@ private void initializeLoader(StoreConfiguration cfg, CacheLoader loader, Initia
          storesMutex.writeLock().lock();
          try {
             loaders.add(loader);
+
+            long interval = configuration.persistence().availabilityInterval();
+            ConnectableObservable<StoreAvailability> observable = Observable.interval(interval, TimeUnit.MILLISECONDS, Schedulers.from(persistenceExecutor))
+                  .map(t -> new StoreAvailability(loader, loader.isAvailable()))
+                  .doOnError(err -> log.debugf("Error encountered when calling isAvailable on %s: %s", loader, err))
```

Comment:
Maybe we want to do http://reactivex.io/RxJava/2.x/javadoc/io/reactivex/Observable.html#onErrorReturn-io.reactivex.functions.Function- and print the error and change it to a not available status `new StoreAvailability(loader, false)`?